### ISSUE TITLE
Embed Bot into Sharepoint 2013 on prem with directline inte

### ISSUE THREAD

**Issue by vishnu2017** [Microsoft/BotFramework-WebChat#1722]
Hi,

I am developing a Bot for a customer using v4 SDK. My end goal is to deploy this BOT on to a sharepoint 2013 environment.

I get this wierd error when I try to deploy stuff into a sharepoint site.

![image](https://user-images.githubusercontent.com/20475543/52719901-91f38400-2fcc-11e9-99ae-8f476a37995f.png)

This works on a normal web application but fails when deployed on a sharepoint site.

Error : As per the screenshot it says its failing at fillOpacity which is a method from botchat.js

Note : I have tried botchat-es5.js webchat-es5.js but have no luck

Can I get any help on this? Thanks for all the good work.

**Comment by corinagum** [comment]
Hi @vishnu2017 - I had this issue transferred over from BotBuilder since this is a Web Chat question. 
Could you clarify a few questions for me? 

- It looks like you are using v4 syntax, but you mention that you are using `botchat.js` (and have tried `botchat-es5` and `webchat-es5`). `webchat.js` indicates use of the v4 CDN - could you let me know which version of Web Chat you intend to use? We might need to do a pass to verify you are using the correct CDN of your intended version.
  - If you are using v4, we have a bunch of [samples ](https://github.com/Microsoft/BotFramework-WebChat/tree/master/samples) that can help you get started.
- What version of IE is being used? Web Chat only supports IE11.
- Could you let me know what your console errors say? I see in your screencap that there are 2.

it looks like `fillOpacity` is from one of our dependencies, `cssstyle`. Hopefully this will help us track down the problem.

**Comment by vishnu2017** [comment]
Hey Corinagum,

Thanks for the quick response.

Firstly, here's my code just for reference.

```
<link rel="stylesheet" href="~/css/botchat.css" /> --- **I have a local copy of the cdn latest versions**
<script src="~/js/botchat.js"></script> --- **I have a local copy of the cdn latest versions**
<!-- Bot Div -->
<div id="chatWindow">
    <div id="bot"></div>
</div>

<!-- Script for DirectLine -->
<script>
    var botConnection = new BotChat.DirectLine({
        token: '------------Passing the token--------------',
    });

    BotChat.App({
        botConnection: botConnection,
        user: { id: 'userid', name: 'User' },
        bot: { id: 'botid' },
        resize: 'detect'
    }, document.getElementById("bot"));

    $('.BOTHEADER).replaceWith('<img src="/images/Logo.png" style="width: 20px;height: 20px;background: none;margin-right: 5px;margin-bottom: 3px"/><span style="display:inline-block; width:"30" ;"></span><b>NAME</b>');

</script>
```

So I use the same logic in a MVC application and everything works as expected.

When I move the same to embed it on a web page of sharepoint 2013 on prem environment, it pops out with an error.



So just wanted to understand, Is botchat.js compatible with SP2013?

Answering your questions :
1. Latest version of botchat.js file targeting v4.
2. I am on IE 11.
![image](https://user-images.githubusercontent.com/20475543/52748429-832cc180-300c-11e9-82bf-c20b1bdc9ccb.png)

3. console error
![image](https://user-images.githubusercontent.com/20475543/52748224-f6820380-300b-11e9-8960-2a63563c7f3f.png)


Here is the bot UI on a MVC application
![image](https://user-images.githubusercontent.com/20475543/52748542-c8e98a00-300c-11e9-9305-c32b84d8d62b.png)

**Comment by corinagum** [comment]
It looks like your intent is to use v3 of Web Chat, not v4. Your screencap of the MVC application and your code indicate that to me. 
First, I would make sure that your copy of the CDN is the correct file -- if you are using `webchat.js`, you are likely using the incorrect file. You should be using a `botchat.js` file, unless you have renamed it. (There are significant differences between v4 and v3, and the CDN's are not interchangeable)

I suspect this will resolve the error of 'BotChat' being undefined, since the `BotChat` object is from v3 and no longer used in v4.

Please see the [index.html](https://github.com/Microsoft/BotFramework-WebChat/blob/master/samples/webchat-v3/index.html) of our v3 sample. Hopefully this will help you get set up. 

Unfortunately I am not sure about Web Chat working on SharePoint -- as far as I know it should, and looking through our repo didn't reveal any issues between v3 and SharePoint. Please let us know what you find out!

**Comment by vishnu2017** [comment]
Hi Corinagum,

I actually bumped into this link [https://github.com/Microsoft/BotFramework-WebChat/issues/1326](https://github.com/Microsoft/BotFramework-WebChat/issues/1326)

I have used v4 logic and here's my code-
`<!DOCTYPE html>
<html lang="en-US">
  <head>
    <title>Web Chat: Full-featured bundle with ES5 polyfills</title>
    
    <script src="https://cdn.botframework.com/botframework-webchat/master/webchat-es5.js"></script>
    <style>
      html, body { height: 100% }
      body { margin: 0 }

      #webchat {
        height: 100%;
        width: 100%;
      }
    </style>
  </head>
  <body>
    <div id="webchat" role="main"></div>
    <script>
       function () {
       
		var tokenval = 'kyTrsRHi3xs.dAA.T---DummyToken-----EQAWABpAGIAZwBsAGIATQBWAEMAbgA.xxYslFvE1AE.4lmzSR4yEhg.cKV3VMUX2uUm7jY2DqzQJxhrLj6WagjEng7PL97vG98';
		
		
		window.WebChat.renderWebChat({directLine:window.WebChat.createDirectLine({token:tokenval})},document.getElementById('webchat'));
        document.querySelector('#webchat > *').focus();
      }
        
    </script>
  </body>
</html>`



Got this error -

![image](https://user-images.githubusercontent.com/20475543/52784208-5616f800-307a-11e9-9550-670ed81698bb.png)

Some compatibility issue with IE 11. I suspect its the same Fileopacity undefined error which I have mentioned in my first comment.

**I am from Microsoft too, we can chat offline if required. Here's my email videepth@microsoft.com**

**Comment by vishnu2017** [comment]
n.n=function(e){**var t=e&&e.__esModule?function(){return e.default**} --- Column 1028 as per the above screenshot.

@corinagum 

It is breaking at this return statement, not sure what's going wrong.

**Comment by corinagum** [comment]
@vishnu2017 it looks like you have errors in your script. the `Expected identifier` error appears to be related to line 23 of your code.

**Comment by vishnu2017** [comment]
@corinagum my bad. Thanks for pointing it out, I forgot to paste in the function name while I was moving stuff.

[@rysantos](https://github.com/rysantos)

![image](https://user-images.githubusercontent.com/20475543/52815470-c8142f00-30c4-11e9-8808-9dedec21de06.png)

![image](https://user-images.githubusercontent.com/20475543/52815613-23deb800-30c5-11e9-9c54-abd1de549994.png)

![image](https://user-images.githubusercontent.com/20475543/52815631-32c56a80-30c5-11e9-9b11-f6077f3c9e53.png)

**Comment by corinagum** [comment]
@vishnu2017 I just messaged you on Teams -- let's meet over a screen share and chat in real time. I think this issue would be more quickly solved that way. :)

**Comment by corinagum** [comment]
@vishnu2017 Hi Vishnu, I'm going to close this issue since we have it figured out offline. :)

To summarize our offline findings: after making sure the project was using the es-5 version of Web Chat, we added a prettified version of the .js file to debug. Sharepoint was breaking when hitting Adaptive Cards, related to https://github.com/Microsoft/BotFramework-WebChat/issues/1510, which is a fix that will be available in  `/latest/` bits on our 4.3 release. Our temporary workaround in the meantime is to use `/master/` bits until our next release.
